src/test/java/com/google/devtools/build/lib/skyframe/SourceDirectoryIntegrationTest.java

```diff
@@ -228,24 +237,12 @@ public void danglingSymlinkModified_invalidatesAction() throws Exception {
   @Test
   public void crossingPackageBoundary_fails() throws Exception {
     createEmptyFile(sourceDir.getRelative("subdir/BUILD"));
-    // TODO(#25834): This should not crash Bazel.
-    assertThrows(IllegalStateException.class, () -> buildTarget("//foo"));
-    BugReport.getAndResetLastCrashingThrowableIfInTest();
+    assertThrows(BuildFailedException.class, () -> buildTarget("//foo"));
     assertContainsEvent(
         "Directory artifact foo/dir crosses package boundary into package rooted at"
             + " foo/dir/subdir");
   }
 
-  @Test
-  public void infiniteSymlinkExpansion_fails() throws Exception {
-    Path dir = sourceDir.getRelative("subdir/nested2");
-    dir.delete();
-    dir.createSymbolicLink(PathFragment.create(".."));
-    assertThrows(BuildFailedException.class, () -> buildTarget("//foo"));
-    assertContainsEvent("infinite symlink expansion detected");
-    assertContainsEvent("foo/dir/subdir/nested2");
-  }
```

Comment:
Why was this test deleted?

Reply:
Sorry, I'm still fixing merge conflicts. Is it possible that the test doesn't have a BUILD file entry on master? Will ping you when I've figured this out.